fleetctl:test: add a new test for submit command

dongsupark opened the pull request:
Introduce a new test for the fleetctl submit command,
not only for simple units, but also for template units.

NOTE: this PR depends on #1466, which is still pending.

dongsupark commented:
I updated the tree. Changes are:
* in ``doSubmitUnits()``, check whether each input unit exists in the submitted unit.
* remove a redundant flag setting ``sharedFlags.NoBlock``.

tixxdz commented:
lgtm, @jonboulle  https://github.com/coreos/fleet/pull/1466 then this one. Thanks!

jonboulle commented:
Please rebase

dongsupark commented:
@jonboulle rebased.

jonboulle commented on fleetctl/submit_test.go:
```diff
@@ -0,0 +1,140 @@
+			0,
+		},
+		{
+			"submit same unit from an available template",
```
Might as well add some cases that mix non-available/available, and same units, etc.

dongsupark replied:
@jonboulle Thanks, I'll do.

dongsupark commented:
Rebased and updated.
As @jonboulle suggested, the following tests were added:
* submit same unit multiple times
* submit same unit with non-available units
* submit units from available and non-available templates
* submit same unit from an available template with units from non-available templates

jonboulle commented:
LGTM, thnaks
